syntax = "proto3";

package tensorflow.serving;
option cc_enable_arenas = true;

import "tensorflow_serving/apis/get_model_metadata.proto";
import "tensorflow_serving/apis/predict.proto";

// open source marker; do not remove
// PredictionService provides access to machine-learned models loaded by
// model_servers.
service PredictionService {
  // Predict -- provides access to loaded TensorFlow model.
  rpc Predict(PredictRequest) returns (PredictResponse);

  // GetModelMetadata - provides access to metadata for loaded models.
  rpc GetModelMetadata(GetModelMetadataRequest)
      returns (GetModelMetadataResponse);
}
